### PR TITLE
Fix bunch of bugs

### PR DESCRIPTION
**Pull request by panleone**
## Abstract

Fix many bugs that I have accidentally added in the latest PRs

First commit: cherry-pick Duddino's commit from #354  that fixes the bug  ".isLocked is not a function"

Second commit: due to bad merge `getLatestBlocks` was returning early on non-shield wallets

Third commit: make sure that the `blockheight` is updated when toggling network

Fourth and fifth commits: The mempool function `ownTransaction` was broken... So I fixed it, moved the implementation to the wallet class and removed the useless functions `getOwnedVin` and `getOwnedVout`

---

## Testing

Test that you can create txs and the wallet syncs new transactions when new block arrives

### CODE REVIEW

**Review by JSKitty — APPROVED** [review]
**tACK** 080ff07b33de882082f9b79e97f85323e2e4eba5 - wallet is actually usable again

File: scripts/mempool.js [hunk 1]
```diff
@@ -156,18 +115,33 @@ export class Mempool {
...
+            .filter(
+                (_, i) =>
+                    this.getOutpointStatus(
+                        new COutpoint({
+                            txid,
+                            n: i,
+                        })
+                    ) & OutpointState.OURS
+            )
+            .reduce((acc, u) => acc + u?.value ?? 0, 0);
```
[the remark this answers is not in the document]
Comment (panleone) [reply scripts/mempool.js:144]: I don't think they will have any applications outside `getDebit` and `getCredit`, that's why I removed them.
I'd say we re-add them if they become useful again?

[the remark this answers is not in the document]
Comment (JSKitty) [reply scripts/mempool.js:144]: **A-OK** with me, was just my personal comment, no blocker.

**Review by JSKitty — APPROVED** [review]
**tACK** 93024c22ebf3c06fa5130a0b078d245c753cb5bf

Tested a few random TXs and "rolling" Synchronisation quickly, still looks good.

**Pull request merged**